src/main/java/net/sf/marineapi/nmea/io/AbstractDataReader.java

```diff
@@ -91,6 +91,8 @@ public void run() {
 				} else if (!SentenceValidator.isSentence(data)) {
 					parent.fireDataEvent(data);
 				}
+			} catch (IllegalArgumentException iae) {
+				parent.handleException("No registered parser", iae);
```

Comment:
The parsers may also throw IAE, so maybe a specific exception could serve a purpose here, e.g. `UnsupportedSentenceException`. Also, I think there isn't much the parent could do in this situation as any additional parsers should be registered before starting the reader, so maybe just let it pass or log a warn level message.